### ISSUE TITLE
[BUG]The loss does not decrease during the training

### ISSUE THREAD

**Issue by w066650** [NVIDIA/recsys-examples#165]
**Describe the bug**
in sharding.py, make_optimizer_and_shard function, if use apply_megatron_ddp,  the loss does not decrease during the training;  but if replaced this functionuse， and use "from torch.nn.parallel import DistributedDataParallel as DDP",  the loss can decrease.

Please help analyze the reasons，thank you。

**Comment by w066650** [comment]
> which dataset are your using?

tf dataset ；  Some network structures were modified.

**Comment by JacoCheung** [comment]
Hi @w066650, it seems like you are using float16. Does the situation persist if you switch to bfloat16? 

BTW Could you check if the gradient value numerically right? 

Mcore DDP may reject gradient reduce if there is any abnormal value. Generally, float16 should be used together with scaling factor. We have not fully tested the convergency of float16.

**Comment by shijieliu** [comment]
close as can not reproduce.
